### ISSUE TITLE
Tooltip accessibility

### ISSUE THREAD

**Issue by yatil** [w3c/wcag-em-report-tool#190]
I have tested the implementation of the tool tips and it is quite good but not perfect. As the (i) icons are first in flow they get read before entering the field and they are not unique, so [as you can see in the video](http://misc.yatil.fastmail.fm/wcag-em-jaws.mp4) it says [Info expand] → tab →  [Evaluation Commisioner], for example.

Doing this right is incredibly complicated as you’d usually want to enter the field first (to announce the label) and then give the user a possibility to access further information (for example using
aria-describedby). But this doesn‘t help other users so we’d need the (i) icon for them and also for people using screen readers and visually access the page, so we can’t hide it from screen readers which would have other negative side effects.

For a quick fix, I’d recommend to have the label of the field in the icon text, for example:

Instead of: `<span class="sr-only ng-binding">Info expand</span>` use: `<span class="sr-only ng-binding">Expand info for the field “Evaluation Commisioner”</span>`

(Maybe even “for the following field Evaluation Commisioner” to make clear that this is a field
that comes up next.)

I realize that no one of our screen reader users picked that up and I wonder if they are just used to it or they navigate forms differently?!

[Note for future: (i) icons should be buttons instead of links, imho.]

**Comment by WilcoFiers** [comment]
I'm a bit stuck on this info icon issue. The solution you suggest would work I think but it means we create a difference between the info buttons on labels and those on headings, for which this extra text seems redundant. I'm not a fan of this inconsistency. What about adding a title to the anchor element which gives the name of the component, both for the label and the heading? Since this is contextual information, I think a title would be appropriate.

**Comment by yatil** [comment]
Another way could be to use aria-descibedby? But I’m also happy with a title and see what users have to say.

**Comment by WilcoFiers** [comment]
So I've added titles. I've also looked at aria-descibedby but it seems to me these lengthy texts for each fields are a bit much as a description, as they are repeated every time you focus on the field. If we get any comments on this later on we can see about improving this later on. Please confirm this solution has resolved the issue.

**Comment by nitedog** [comment]
I agree that [ Expand info for the following field “Evaluation Commisioner” ] is really long and annoying. I think better would be to move the icon into the label, and call it "show instructions", which is more accurate than "info expand" or "expand info". Not sure if title is well supported?

**Comment by yatil** [comment]
Well, the issue is that even “show instructions” is not telling which instructions for which field as the [icons are out of context](http://misc.yatil.fastmail.fm/wcag-em-jaws.mp4). I haven’t succeeded in getting the title read, but in theory this should work. (Probably need to ask someone who is using a screen reader and knows all the tricks ;-) – That said, I think once using the tool people can figure out which info button belongs to which input field, probably an (invisible) instruction text might help.

I don’t think this is a major issue.

**Comment by nitedog** [comment]
@yatil, I think "[label] - Evaluation Commissioner - [button/link] show instructions" is a good compromise between context and redundancy, keeping in mind that typical users of this tool are more experienced accessibility evaluators. Having said that, yes, user/usability testing is needed and already scheduled.

@WilcoFiers - after you implement these changes, please *don't* close this issue but add the label "question" and move it to milestone "version 2", so that we find this issue again for usability testing.

**Comment by WilcoFiers** [comment]
I've added some aria-expanded stuff to improve the functionality. I don't think there is any point of confusion about these icons anymore.
